WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/FollowersUseCase.kt

```diff
@@ -25,36 +26,52 @@ import org.wordpress.android.ui.stats.refresh.lists.sections.BlockListItem.ListI
 import org.wordpress.android.ui.stats.refresh.lists.sections.BlockListItem.NavigationAction
 import org.wordpress.android.ui.stats.refresh.lists.sections.BlockListItem.TabsItem
 import org.wordpress.android.ui.stats.refresh.lists.sections.BlockListItem.Title
+import org.wordpress.android.ui.stats.refresh.lists.sections.insights.InsightUseCaseFactory
 import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
 import org.wordpress.android.viewmodel.ResourceProvider
 import javax.inject.Inject
 import javax.inject.Named
 
-private const val PAGE_SIZE = 6
+private const val BLOCK_PAGE_SIZE = 6
+private const val VIEW_ALL_PAGE_SIZE = 10
 
 class FollowersUseCase
 @Inject constructor(
     @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
     private val insightsStore: InsightsStore,
     private val statsUtilsWrapper: StatsUtilsWrapper,
     private val resourceProvider: ResourceProvider,
-    private val analyticsTracker: AnalyticsTrackerWrapper
+    private val analyticsTracker: AnalyticsTrackerWrapper,
+    private val isViewAllMode: Boolean
 ) : StatefulUseCase<Pair<FollowersModel, FollowersModel>, Int>(
         FOLLOWERS,
         mainDispatcher,
         0
 ) {
+    private val pageSize = if (isViewAllMode) VIEW_ALL_PAGE_SIZE else BLOCK_PAGE_SIZE
+    private lateinit var lastSite: SiteModel
+
     override suspend fun loadCachedData(site: SiteModel) {
-        val wpComFollowers = insightsStore.getWpComFollowers(site, PAGE_SIZE)
-        val emailFollowers = insightsStore.getEmailFollowers(site, PAGE_SIZE)
+        lastSite = site
+        val wpComFollowers = insightsStore.getWpComFollowers(site, pageSize)
```

Comment:
I'm wondering if this will always work as expected. I think we should load all the cached data when we're opening the `ViewAll` screen. What about making the pageSize nullable and pass null when we want all the items in the DB? I think it makes sense to load everything we have.